src/main/java/com/google/cloud/tools/maven/CloudSdkMojo.java

```diff
@@ -36,31 +35,19 @@
   @Parameter(property = "cloudSdkPath", required = false)
   protected File cloudSdkPath;
 
-  protected CloudSdk.Builder cloudSdkBuilder;
-
-  private final ProcessOutputLineListener gcloudOutputListener = new ProcessOutputLineListener() {
-    @Override
-    public void outputLine(String line) {
-      getLog().info("GCLOUD: " + line);
-    }
-  };
-
   @Parameter(defaultValue = "${pluginDescriptor}", readonly = true)
-  private PluginDescriptor pluginDescriptor;
-
-  protected CloudSdkMojo() {
-    super();
+  protected PluginDescriptor pluginDescriptor;
 
-    cloudSdkBuilder = new CloudSdk.Builder()
-        .addStdOutLineListener(gcloudOutputListener)
-        .addStdErrLineListener(gcloudOutputListener);
-  }
+  protected AppEngineDependenciesFactory factory = new AppEngineDependenciesFactory();
 
-  protected CloudSdk getCloudSdk() {
+  protected CloudSdk.Builder configureCloudSdkBuilder(CloudSdk.Builder cloudSdkBuilder) {
     return cloudSdkBuilder
         .sdkPath(cloudSdkPath)
+        .addStdOutLineListener(factory.gcloudOutputListener(this))
```

Comment:
The factory is being used cyclically here isn't it?

sdk (factory)
factory.deploy(sdk) ?